src/components/TokenSelect/index.tsx

```diff
@@ -129,7 +131,20 @@ export default memo(function TokenSelect({ value, collapsed, disabled, onSelect
   usePrefetchBalances()
 
   const [open, setOpen] = useState(false)
-  const onOpen = useCallback(() => setOpen(true), [])
+  const onTokenSelectorClick = useAtomValue(onTokenSelectorClickAtom)
+  const onOpen = useCallback(() => {
+    const promise = onTokenSelectorClick?.()
+    if (promise) {
+      return promise
```

Comment:
nit: if you're one-lining L146, then you can one-line this, too:

    return promise.then((open) => setOpen(open)).catch(() => setOpen(false))